src/App.tsx

```diff
@@ -1,223 +1,159 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState, useCallback
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoList } from './components/TodoList';
+import { FilterTypes } from './types/FilterTypes';
+import { Footer } from './components/Footer';
+import { TodoField } from './components/TodoField';
+
+import { getTodos, newTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.all);
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedIds, setdeletedIds] = useState<number[]>([]);
+  const [todoToAdd, setTodoToAdd] = useState('');
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case FilterTypes.active:
+        return !todo.completed;
+
+      case FilterTypes.completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos(user?.id || 0);
+
+        setTodos(loadedTodos);
+      } catch {
+        setError('load');
+        setTimeout(() => setError(''), 3000);
```

Comment:
To avoid doing it by hand every time, let's `useEffect` work instead of us. You can write useEffect with `error` in the dependency array. And check: if the error is not empty, set it empty after 3000 ms as you do.